src/__tests__/Document.test.ts

```diff
@@ -856,7 +856,46 @@ describe('validate', () => {
 			});
 			const document = new DocumentSubclass(schema, { data: { prop1: 'foo' } });
 
-			const expected = new Map([['prop1', 'Test error message']]);
+			const expected = new Map([['prop1', ['Test error message']]]);
+			expect(await document.validate()).toEqual(expected);
+		});
+
+		test('should return error at property & unravel nested errors if schemaType validation fails', async () => {
```

Comment:
This test is missing coverage of setting a schema to a property (effectively as an embedded object) and setting a schema as an object-array definition.

I think that might be what you're trying to do with the `nestedSchema` and `subDocumentArraySchema` but those are just objects and not instances of `Schema`.

It looks like you've covered the non-schema version of an embedded object with the `nestedObject` structure, so switching `nestedSchema` to be a `Schema` instance will cover both directly embedding and having a schema. You don't have two versions of a subdocument array, so you'd need to exercise both.

I think you should be careful with the wording as you've got a `nestedObject` property and then an `embeddedType` which uses something called `nestedSchema`. That gets a bit confusing.